src/app/hooks/useOptimizelyVariation/index.jsx

```diff
@@ -21,4 +30,7 @@ const useOptimizelyVariation = (experimentId, overrideAttributes = {}) => {
   return variation;
 };
 
+const useOptimizelyVariation = (useClientSide = isClientSide) =>
+  useClientSide ? useClientSideOptimizelyVariation : useTrueFunction;
+
```

Comment:
Not seen a curry function used for hooks, a nice way to get round the rule of hooks. However, curried functions can cause a bit of confusion, especially with reacts hook `use` pattern. Because we're hard coding the `isClientSide` value, I'd suggest going against the rule for this specific use case by using a conditional within the hook and disabling the linting rule for that line, as the state will always be persistent across renders ([rules of hooks](https://reactjs.org/docs/hooks-rules.html#explanation)) because `isClientSide` is always false. Worth adding a comment if you decide to change the method to this.